tdrs-backend/tdpservice/search_indexes/management/commands/clean_and_reparse.py

```diff
@@ -0,0 +1,236 @@
+                qset = model.objects.all().filter(datafile_id__in=file_ids)
+                total_deleted += qset.count()
+                if not new_indices:
+                    # If we aren't creating new indices, then we don't want duplicate data in the existing indices.
+                    doc().update(qset, refresh=True, action='delete')
+                qset._raw_delete(qset.db)
+            except BulkIndexError as e:
+                log(f'Elastic document delete failed for model {model}. Database and Elastic are INCONSISTENT! Restore '
+                    'the DB from the backup as soon as possible!',
+                    logger_context=log_context,
+                    level='critical')
+                raise e
+            except Exception as e:
+                log(f'_raw_delete failed for model {model}. Database and Elastic are INCONSISTENT! Restore the DB from '
+                    'the backup as soon as possible!',
+                    logger_context=log_context,
+                    level='critical')
+                raise e
+        return total_deleted
+
+    def __handle_datafiles(self, files, log_context):
+        """Delete, re-save, and re-parse selected datafiles."""
+        for f in files:
+            try:
+                f.delete()
+            except Exception as e:
+                log(f'DataFile.delete failed for id: {f.pk}. Database and Elastic are INCONSISTENT! Restore the '
+                    'DB from the backup as soon as possible!',
+                    logger_context=log_context,
+                    level='critical')
+                raise e
+
+            try:
+                f.save()
+            except Exception as e:
+                log(f'DataFile.save failed for id: {f.pk}. Database and Elastic are INCONSISTENT! Restore the '
+                    'DB from the backup as soon as possible!',
+                    logger_context=log_context,
+                    level='critical')
+                raise e
+
+            # latest version only? -> possible new ticket
+            parser_task.parse.delay(f.pk, should_send_submission_email=False)
+
+    def handle(self, *args, **options):
+        """Delete and re-parse datafiles matching a query."""
+        fiscal_year = options.get('fiscal_year', None)
+        fiscal_quarter = options.get('fiscal_quarter', None)
+        delete_all = options.get('all', False)
+        new_indices = options.get('new_indices', False)
+        delete_indices = options.get('delete_indices', False)
+
+        backup_file_name = "/tmp/reparsing_backup"
+        files = None
+        continue_msg = "You have selected to re-parse datafiles for FY {fy} and {q}. The re-parsed files "
+        if delete_all:
+            files = DataFile.objects.all()
+            backup_file_name += "_FY_All_Q1-4"
+            continue_msg = continue_msg.format(fy="All", q="Q1-4")
+        else:
+            if not fiscal_year and not fiscal_quarter:
+                print(
```

Comment:
`print` intended vs logger?

Reply:
Same as above. It needs to be std::out to indicate to the user that something is wrong, not to the developer.